internal/generator/tls.go

```diff
@@ -0,0 +1,39 @@
+package generator
+
+import (
+	configv1 "github.com/openshift/api/config/v1"
+	logging "github.com/openshift/cluster-logging-operator/apis/logging/v1"
+	"github.com/openshift/cluster-logging-operator/internal/tls"
+	"strings"
+)
+
+// TLSProfileInfo returns the minTLSVersion, ciphers as a comma-separated list given the available TLSSecurityProfile
+func (op Options) TLSProfileInfo(clfSpec *configv1.TLSSecurityProfile, outputSpec logging.OutputSpec) (string, string) {
+
+	var tlsProfileSpec *configv1.TLSProfileSpec
+	if _, ok := op[ClusterTLSProfileSpec]; ok {
+		clusterSpec := op[ClusterTLSProfileSpec].(configv1.TLSProfileSpec)
+		tlsProfileSpec = &clusterSpec
+	}
+	profiles := []*configv1.TLSSecurityProfile{clfSpec}
+	if outputSpec.TLS != nil {
+		profiles = append(profiles, outputSpec.TLS.TLSSecurityProfile)
+	}
+	for _, profile := range profiles {
+		if profile != nil {
+			if profile.Type == configv1.TLSProfileCustomType {
+				tlsProfileSpec = &profile.Custom.TLSProfileSpec
+			} else {
+				tlsProfileSpec = configv1.TLSProfiles[profile.Type]
+			}
+		}
+
+	}
```

Comment:
Just IMO, it'd be easier to follow if the logic flow was schematically like this:
`
if outputSpec.TLS != nil { 
     tlsProfileSpec = configv1.TLSProfiles[outputSpec.TLS.TLSSecurityProfile.Type]
} else if clfSpec != nil {
     tlsProfileSpec = configv1.TLSProfiles[clfSpec.Type]
} else { 
      tlsProfileSpec =  &op[ClusterTLSProfileSpec].(configv1.TLSProfileSpec)
}
`